### ISSUE TITLE
Bug: UselessVariable causes syntax error with assignment after assignment

### ISSUE THREAD

**Issue by Levivb** [slevomat/coding-standard#420]
Hey,

Note: this might be a duplicate of https://github.com/slevomat/coding-standard/issues/419

Last weekend I noticed the following (simplified):
```php
function getTitleAttribute($value = null) {
    $this->title = $value = $this->url ?: $this->somethingElse;

    return $value;
}
```

Which, with the `UselessVariable` sniff is fixed to:
```php
function getTitleAttribute($value = null) {
    $this->title = return $this->url ?: $this->somethingElse;
}
```

Note the return statement after the `$this->title` assignment

This construct here was necessary due to some Laravel magic which prevents the `return $this->title` since it would end up in an infinite loop.

The resulting code does however error out in a rather dramatic fashion: syntax error<br>
For me, the linter in the pipeline catched the error, but not everyone might be so lucky.

I'm not sure what the potential bugfix here would be. One could argue the code sniffer should run a linter over each file it touches, although that should probably be something for phpcs itself. Perhaps skipping this sniff if the variable assignment is preceded by other code (see https://github.com/slevomat/coding-standard/issues/419 as well).

Let me know what you think

Thanks in advance

**Comment by kukulich** [comment]
Thank you for your report.

Should be fixed with https://github.com/slevomat/coding-standard/commit/e470c10c66f4496c723d83fad2ccf7cf0729ee7f
